modules/charon-core/src/test/java/org/wso2/charon3/core/schema/ServerSideValidatorTest.java

```diff
@@ -0,0 +1,687 @@
+                        "urn:ietf:params:scim:schemas:extension:2.0:Role:permissions", "permissions", REFERENCE,
+                        true, "", false, false,
+                        READ_WRITE, DEFAULT, NONE, null, null, null);
+        SCIMResourceTypeSchema roleResourceSchema =
+                SCIMResourceTypeSchema.createSCIMResourceSchema(schemasListRole, attributeSchemaRole1,
+                        attributeSchemaRole2, attributeSchemaRole3, attributeSchemaRole4, attributeSchemaRole5);
+
+        return  roleResourceSchema;
+    }
+
+    private SCIMResourceTypeSchema createSCIMResourceTypeSchemaCustomUser() {
+
+        List<String> schemasList1 = new ArrayList<>();
+        schemasList1.add("urn:ietf:params:scim:schemas:core:2.0:User");
+        schemasList1.add("urn:ietf:params:scim:schemas:extension:enterprise:2.0:User");
+        AttributeSchema subAttributeSchemaValue =
+                SCIMAttributeSchema.createSCIMAttributeSchema(
+                        "urn:ietf:params:scim:schemas:core:2.0:User:emails.value",
+                        "value", STRING, false, "", false, false,
+                        READ_WRITE, DEFAULT, NONE, null, null, null);
+        ArrayList<AttributeSchema> subAttributeSchemaListEmails = new ArrayList<>();
+        subAttributeSchemaListEmails.add(subAttributeSchemaValue);
+        AttributeSchema attributeSchemaCustom =
+                SCIMAttributeSchema.createSCIMAttributeSchema(
+                        "urn:ietf:params:scim:schemas:extension:2.0:CustomResource:attribute1",
+                        "attribute1", STRING, true, "", true, false,
+                        READ_WRITE, DEFAULT, NONE, null, null, null);
+        AttributeSchema attributeSchemaEmail =
+                SCIMAttributeSchema.createSCIMAttributeSchema(
+                        "urn:ietf:params:scim:schemas:core:2.0:User:emails",
+                        "emails", COMPLEX, true, "", false, false,
+                        READ_WRITE, DEFAULT, NONE, null, null, subAttributeSchemaListEmails);
+        SCIMResourceTypeSchema scimResourceTypeSchema1 = SCIMResourceTypeSchema.createSCIMResourceSchema(
+                schemasList1, attributeSchemaCustom, attributeSchemaEmail);
+
+        return scimResourceTypeSchema1;
+    }
+
+    @DataProvider(name = "dataForValidateCreatedSCIMObjectSuccess")
+    public Object[][] dataToValidateCreatedSCIMObjectSuccess() throws InstantiationException, IllegalAccessException {
+
+        User user = createNewUser();
+        Group group = createNewGroup();
+        Role role = createNewRole();
+
+        SCIMResourceTypeSchema userResourceSchema = createSCIMResourceTypeSchemaUser();
+        SCIMResourceTypeSchema groupResourceSchema = createSCIMResourceTypeSchemaGroup();
+        SCIMResourceTypeSchema roleResourceSchema = createSCIMResourceTypeSchemaRole();
+
+        return new Object[][]{
+
+                {user, userResourceSchema},
```

Comment:
Since there is no spacial usage of the variables, let's remove the variables and call the methods directly from here.